tests/blackbox/src/common/metrics.py

```diff
@@ -0,0 +1,66 @@
+class Metrics:
+    """Singleton class to store metrics."""
+
+    __instance = None
+    init_conversation_response_times_sec: list[float] = []
+    conversation_response_times_sec: list[float] = []
+
+    @staticmethod
+    def get_instance():
+        """Static access method."""
+        if Metrics.__instance is None:
+            Metrics()
+        return Metrics.__instance
+
+    def __init__(self):
+        """Virtually private constructor."""
+        if Metrics.__instance is not None:
+            raise Exception("This class is a singleton!")
+        else:
+            Metrics.__instance = self
```

Comment:
Q: Can you explain this design?

Reply:
https://refactoring.guru/design-patterns/singleton